R/lockfile.R

```diff
@@ -184,13 +184,11 @@ renv_lockfile_create <- function(project,
       packages = packages,
       exclude  = exclude
     ),
-    renv_recompute_records = function(cnd) {
-      NULL
-    }
+    renv_recompute_records = identity
   )
 
   # We re-computing and re-filtering records once, if requested by user
-  if (is.null(records)) {
+  if (inherits(records, "renv_recompute_records")) {
```

Comment:
Also it's not directly related to the goal of this PR?

Reply:
I think my main concern about a sentinel value like `NULL` is that functions might reasonably return `NULL` in some cases, so it's hard to reason that the `NULL` was definitely produced because of the condition.

Either way though I think I landed on something we'll both agree is better?